### PR TITLE
Add basic network stuff

### PR DESCRIPTION
**Pull request by huntears**
- [x] Accept clients
- [x] Receive client data
- [x] Send data to clients

### CODE REVIEW

**Review by spectrapulse — COMMENTED** [review]
I have no idea what I'm reading


lgtm!

File: cubic-server/Client.hpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#ifndef D23DD5CC_1F28_49BB_B77D_E244C60CC705
+#define D23DD5CC_1F28_49BB_B77D_E244C60CC705
```
Comment (Trompettesib) [inline cubic-server/Client.hpp:2]: not sure about that but why not ^^

File: cubic-server/Server.hpp [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#ifndef F43D56DD_C750_470F_A7C9_27CE21D37FC3
+#define F43D56DD_C750_470F_A7C9_27CE21D37FC3
```
Comment (Trompettesib) [inline cubic-server/Server.hpp:2]: not sure about that too

File: cubic-server/main.cpp [hunk 3]
```diff
@@ -1,357 +1,21 @@
...
-
-    for (; list; list = list->next)
-    {
-        if (list->cli.to_remove)
-        {
-            remove_client(srv, list);
-            return;
-        }
-    }
+    std::cout << "Usage:\n\t" << caller << std::endl;
```
Comment (Trompettesib) [inline cubic-server/main.cpp:8]: maybe not enough understandable

**Review by Trompettesib — APPROVED** [review]
everything look good to me ! maybe some changes to do about the "-h"

File: cubic-server/main.cpp (hunk 3 above)
Comment (huntears) [reply cubic-server/main.cpp:8]: It's only temporary

**Pull request merged**